src/controllers/controller.cpp

```diff
@@ -2,14 +2,27 @@
 
 #include <QApplication>
 #include <QJSValue>
+#include <QRegularExpression>
 
 #include "controllers/controllerdebug.h"
 #include "controllers/defs_controllers.h"
 #include "moc_controller.cpp"
 #include "util/screensaver.h"
 
-Controller::Controller()
-        : m_pScriptEngineLegacy(nullptr),
+namespace {
+const QRegularExpression invalidLoggingCategoryChars("[^a-z0-9]");
```

Comment:
This constant along with some utility functions belongs to RuntimeLoggingCategory which currently accepts arbitrary strings without any validation.

Reply:
...or are these restrictions specific for this particular use case and in general arbitrary strings are allowed?

Reply:
good idea

Reply:
The restrictions are not specific to this use case:

> Category names are free text; to configure categories using Logging Rules, their names should follow this convention:
> Use letters and numbers only.
> Use dots to further structure categories into common areas.
> Avoid the category names: debug, info, warning, and critical.
> Category names with the qt prefix are solely reserved for Qt modules.

https://doc.qt.io/qt-5/qloggingcategory.html#creating-category-objects